pkg/operator/controller/status.go

```diff
@@ -332,8 +332,8 @@ func computeOperatorAvailableCondition(oldCondition *configv1.ClusterOperatorSta
 		availableCondition.Message = "At least 1 DNS DaemonSet available"
 	} else {
 		availableCondition.Status = configv1.ConditionFalse
-		availableCondition.Reason = "DNSUnavailable"
-		availableCondition.Message = "No DNS DaemonSets available"
+		availableCondition.Reason = "NoDaemonSetsAvailable"
+		availableCondition.Message = "No DNS DaemonSets are available"
```

Comment:
The reason and message may be misleading.  The problem is that no DNS resource is available.  The default DNS resource may be unavailable because the Service does not have an address assigned even while the DaemonSet is available.

Reply:
Since the DNS is a singleton, should the operator availability reason mirror the DNS reason in this case? @bparees

Reply:
Seems plausible to me....

Reply:
Likewise for the Degraded condition?

Reply:
yeah i think so

Reply:
Unfortunately we need to do a bit of refactoring here to propagate the reason, lifting newer techniques from ingress-operator. Will keep this PR open, but not sure if I'll get back to it in the 4.2 dev cycle.